solvation_analysis/plotting.py

```diff
@@ -0,0 +1,319 @@
+
+def format_graph():
+    # a formatting/styling decorator
+    """
+
+    Parameters
+    ----------
+
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+
+    return
+
+
+def plot_network_size_histogram(networking):
+    """
+    Returns a histogram of network sizes.
+
+    Parameters
+    ----------
+    networking : Networking
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    network_sizes = networking.network_sizes
+    sums = network_sizes.sum(axis=0)
+    total_networks = sums.sum()
+    fig = go.Figure()
+    fig.add_trace(go.Bar(x=sums.index, y=sums.values/total_networks))
+    fig.update_layout(xaxis_title_text="Network Size", yaxis_title_text="Fraction of All Networks",
+                      title="Histogram of Network Sizes")
+    fig.update_xaxes(type="category")
+    return fig
+
+
+def plot_shell_size_histogram(solution):
+    """
+    Returns a histogram of shell sizes.
+
+    Parameters
+    ----------
+    solution : Solution
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    # TODO: orion suggests maybe consider having the option to replace the solvent names with
+    # custom solvent names, perhaps via an input dictionary
+    # ideally this same API could be reused throughout the plotting API
+
+    speciation_data = solution.speciation.speciation_data
+    speciation_data["total"] = speciation_data.sum(axis=1)
+    sums = speciation_data.groupby("total").sum()
+    fig = go.Figure()
+    totals = sums.T.sum()
+    for column in sums.columns:
+        fig.add_trace(go.Bar(x=sums.index.values, y=sums[column].values/totals, name=column))
+    fig.update_layout(xaxis_title_text="Shell Size", yaxis_title_text="Fraction of Total Molecules",
+                      title="Fraction of Solvents in Shells of Different Sizes")
+    fig.update_xaxes(type="category")
+    return fig
+
+
+def plot_speciation(solution):
+    # square area
+    # should be doable with plotly.express.imshow and go.add_annotations
+    return
+
+
+def plot_co_occurrence(solution):
+    return
+
+
+def plot_clustering(solution):
+    # not in this branch yet
+    return
+
+
+def plot_coordinating_atoms(solution):
+    # for each solvent
+    # by atom type? could allow by element or other features?
+    # bar chart with one bar for each solvent
+    # normalized
+    return
+
+
+# multiple solutions
+def compare_solvent_dicts(properties, coerce_solvent_names, keep_solvents, legend_label, x_axis="species", series=False):
+    # generalist plotter, this can plot either bar or line charts of the same data
+    """
+
+    Parameters
+    ----------
+    properties : dictionary of the solvent property to be compared
+    coerce_solvent_names : a dictionary where the keys are strings of solvent names and the values are
+        strings of a more generic name for the solvent (i.e. {"EAf" : "EAx", "fEAf" : "EAx"})
+    keep_solvents : a list of strings of solvent names that are common to all systems in question,
```

Comment:
maybe `solvents_to_plot` instead of `keep_solvents`. We should also document what the behavior is if this is not set.